pulsar-broker/src/test/java/org/apache/pulsar/broker/service/ReplicatorSubscriptionTest.java

```diff
@@ -704,6 +705,66 @@ public void testReplicatedSubscriptionWhenReplicatorProducerIsClosed() throws Ex
         Awaitility.await().untilAsserted(() -> assertNotNull(topic2.getSubscription(subscriptionName)));
     }
 
+    @Test
+    public void testReplicateSubBackLog() throws Exception {
+        String namespace = BrokerTestUtil.newUniqueName("pulsar/testReplicateSubBackLog");
+        String topic = "persistent://" + namespace + "/when-replicator-producer-is-closed";
+        String subName = "sub";
+
+        admin1.namespaces().createNamespace(namespace);
+        pulsar1.getConfiguration().setReplicatedSubscriptionsSnapshotFrequencyMillis(100);
+        @Cleanup
+        PulsarClient client1 = PulsarClient.builder().serviceUrl(url1.toString())
+                .statsInterval(0, TimeUnit.SECONDS)
+                .build();
+        @Cleanup
+        Consumer<byte[]> consumer = client1.newConsumer()
+                .topic(topic)
+                .subscriptionName(subName)
+                .ackTimeout(5, TimeUnit.SECONDS)
+                .subscriptionType(SubscriptionType.Shared)
+                .replicateSubscriptionState(true)
+                .subscribe();
+        @Cleanup
+        Producer<byte[]> producer = client1.newProducer()
+                .topic(topic)
+                .create();
+        for (int i = 0; i < 10; i++) {
+            producer.newMessage().send();
+        }
+        TopicStats topicStats = admin1.topics().getStats(topic, false, true);
+        long backlogSize = topicStats.getBacklogSize();
+        //There should no replicator snapshot marker write into the topic, so the backlog would not change.
+        try {
+            Awaitility.await().untilAsserted(() -> {
+                TopicStats stats = admin1.topics().getStats(topic, false, true);
+                assertNotEquals(stats.getBacklogSize(), backlogSize);
+            });
+            fail();
+        } catch (org.awaitility.core.ConditionTimeoutException ex) {
+        }
```

Comment:
You want to confirm no marker was sent after `10s`, right? It is better to verify after `sleep(10s)`.